mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/query/CriterionConverter.java

```diff
@@ -209,6 +209,9 @@ private QueryBuilder convertFlag(SearchQuery.FlagCriterion flagCriterion) {
         if (flag.equals(Flags.Flag.FLAGGED) ) {
             return boolQuery().filter(termQuery(JsonMessageConstants.IS_FLAGGED, operator.isSet()));
         }
+        if (flag.equals(Flags.Flag.USER) ) {
+            return boolQuery();
```

Comment:
Make this consistent across implementation and add a test for it. It's not normal that for ES you say yes for everything, and in Memory you say:

```
         if (flag == Flags.Flag.USER) {
            return isSet == message.createFlags().getUserFlags().length > 0;
         }
```

Please test it as part of the SearchIndex APIs